### PR TITLE
add blackdoc

### PR DESCRIPTION
**Pull request by keewis**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] ~If static libraries are linked in, the license of the static library is packaged.~
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by keewis** [comment]
the `win_64 win` CI fails because of the syntax chosen to set the environment variable (`SETUPTOOLS_SCM_PRETEND_VERSION`). Since the package is `noarch: python` I think this can be ignored?

**Comment by scopatz** [comment]
Yes it can!

File: recipes/blackdoc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  noarch: python
+  entry_points:
+    - blackdoc=blackdoc.__main__:main
+  script: SETUPTOOLS_SCM_PRETEND_VERSION={{ version }} {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
+    - setuptools >=45
```
Comment (scopatz) [inline recipes/blackdoc/meta.yaml:24]: You don't need setuptools since you have pip

Comment (keewis) [reply recipes/blackdoc/meta.yaml]: true, but `pip` doesn't limit the version of `setuptools`. I'm not quite sure why I chose that version, but the source's `pyproject.toml` specifies `setuptools>=42` so the recipe should copy that.

**Comment by keewis** [comment]
thanks, @scopatz

**Comment by keewis** [comment]
it seems there's something wrong with the build: the [build](https://travis-ci.com/github/conda-forge/staged-recipes/builds/186060423) failed and [conda-forge/blackdoc-feedstock](https://github.com/conda-forge/blackdoc-feedstock) is empty. Does anyone know how to fix that?

**Pull request merged**